pkg/apis/clusteroperator/types.go

```diff
@@ -720,3 +720,58 @@ const (
 	// NodeTypeCompute is a node that is a compute node in the cluster
 	NodeTypeCompute NodeType = "Compute"
 )
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DNSZone represents a Domain Name Service Zone to manage
+type DNSZone struct {
+	// +optional
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	// +optional
+	Spec DNSZoneSpec
+	// +optional
+	Status DNSZoneStatus
+}
+
+// DNSZoneSpec is a specification of a Domain Name Service Zone. The specification will
+// be specific to each cloud provider.
+type DNSZoneSpec struct {
+	// Zone is the DNS zone to host
+	DNSZone string
+
+	// AWS specifies cluster hardware configuration on AWS
+	// +optional
+	AWS *AWSDNSZoneSpec
+}
+
+// AWSDNSZoneSpec contains the AWS specific DNSZone specifications
+type AWSDNSZoneSpec struct {
+	// AccountSeceret refers to a secret that contains the AWS account access
+	// credentials
+	AccountSecret corev1.LocalObjectReference
+
+	// Region specifies the AWS region where the cluster will be created
+	Region string
+}
+
+// DNSZoneStatus is the status of a DNSZone. It may be used to indicate if the
+// DNS zone is ready to be used, or if any problems have been detected.
+type DNSZoneStatus struct {
+	// CreationTimestamp is the time that the zone was created.
+	CreationTimestamp *metav1.Time
```

Comment:
I agree, this is confusing. I'll rename it.

Currently this is the time that the controller created the hosted zone in AWS. So it should only be set one time.

I'm fine adding a different status for LastSyncTime, which I think I'll need to do to address one of your other comments.

Or, should I just replace CreationTimestamp with LastSyncTime?

Hosted zone creation time is in AWS, so it may not be a big deal to keep it. Then again, we may want it for other clouds / environments that may not have it.

Thoughts?

Reply:
I'd be inclined to replace it, can't think of many use cases where we'd want to know when it was first created in AWS.